packages/flutter/lib/src/rendering/proxy_box.dart

```diff
@@ -841,7 +840,7 @@ class RenderOpacity extends RenderProxyBox {
       if (_alpha == 255) {
         // No need to keep the layer. We'll create a new one if necessary.
         layer = null;
-        context.paintChild(child, offset);
+        context.paintChild(child!, offset);
         return;
       }
       assert(needsCompositing);
```

Comment:
on the next line, how come `layer` can be cast to `OpacityLayer` rather than `OpacityLayer?`?